cmd/prestop/main.go

```diff
@@ -0,0 +1,231 @@
+	b, err := json.MarshalIndent(i, "", "  ")
+	if err != nil {
+		fmt.Println("error:", err)
+	}
+	fmt.Println(string(b))
+	zap.S().Info(string(b))
+}
+
+// shouldDeletePod returns a boolean value indicating if this pod should be deleted
+// this would be the case if the agent is currently trying to upgrade the version
+// of mongodb.
+func shouldDeletePod(health agenthealth.Health) (bool, error) {
+	hostname := os.Getenv("HOSTNAME")
+	status, ok := health.ProcessPlans[hostname]
+	if !ok {
+		return false, fmt.Errorf("hostname %s was not in the process plans", hostname)
+	}
+	return isWaitingToBeDeleted(status), nil
+}
+
+// getAgentHealthStatus returns an instance of agenthealth.Health read
+// from the health file on disk
+func getAgentHealthStatus() (agenthealth.Health, error) {
+	f, err := os.Open(os.Getenv(agentStatusFilePathEnv))
+	if err != nil {
+		return agenthealth.Health{}, fmt.Errorf("error opening file: %s", err)
+	}
+	defer f.Close()
+
+	h, err := readAgentHealthStatus(f)
+	if err != nil {
+		return agenthealth.Health{}, fmt.Errorf("error reading health status: %s", err)
+	}
+	return h, err
+
+}
+
+// getThisPod returns an instance of corev1.Pod that points to the current pod
+func getThisPod() (corev1.Pod, error) {
+	podName := os.Getenv("HOSTNAME")
+	if podName == "" {
+		return corev1.Pod{}, fmt.Errorf("environment variable HOSTNAME was not present")
+	}
+
+	ns, err := getNamespace()
+	if err != nil {
+		return corev1.Pod{}, fmt.Errorf("error reading namespace: %+v", err)
+	}
+
+	return corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: ns,
+		},
+	}, nil
+}
+
+// readAgentHealthStatus reads an instance of health.Health from the provided
+// io.Reader
+func readAgentHealthStatus(reader io.Reader) (agenthealth.Health, error) {
+	var h agenthealth.Health
+	data, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return h, err
+	}
+	err = json.Unmarshal(data, &h)
+	return h, err
+}
+
+func ensureEnvironmentVariables(requiredEnvVars ...string) error {
+	var missingEnvVars []string
+	for _, envVar := range requiredEnvVars {
+		if val := os.Getenv(envVar); val == "" {
+			missingEnvVars = append(missingEnvVars, envVar)
+		}
+	}
+	if len(missingEnvVars) > 0 {
+		return fmt.Errorf("missing envars: %s", strings.Join(missingEnvVars, ","))
+	}
+	return nil
+}
+
+func main() {
+	fmt.Println("Calling pre-stop hook!")
+	cfg := zap.NewDevelopmentConfig()
+	if err := ensureEnvironmentVariables(logFilePathEnv, agentStatusFilePathEnv); err != nil {
+		zap.S().Fatal("Not all required environment variables are present: %s", err)
+		os.Exit(1)
+	}
+	cfg.OutputPaths = []string{
+		os.Getenv(logFilePathEnv),
+	}
+	log, err := cfg.Build()
+	if err != nil {
+		zap.S().Errorf("Error building logger config: %s", err)
+		os.Exit(1)
+	}
+	logger = log.Sugar()
+	health, err := getAgentHealthStatus()
+	if err != nil {
+		logger.Errorf("Error getting the agent health file: %s", err)
+	}
+
+	prettyPrint(health)
```

Comment:
this was a temporary debugging print, although getting the full health status file at the time this hook was called could still be useful.